OpenRA.Game/Server/Server.cs

```diff
@@ -194,23 +133,135 @@ public Server(IPEndPoint endpoint, ServerSettings settings, ModData modData)
+		public void Shutdown()
+		{
+			State = ServerState.ShuttingDown;
+		}
+
+		public void StartGame()
+		{
+			listener.Stop();
+
+			Console.WriteLine("Game started");
+
+			// Drop any unvalidated clients
+			foreach (var c in preConns.ToList())
+				DropClient(c);
+
+			// Drop any players who are not ready
+			foreach (var c in Connections)
+			{
+				if (GetClient(c).IsInvalid)
+				{
+					SendOrderTo(c, "ServerError", "You have been kicked from the server");
+					DropClient(c);
+				}
+			}
+
+			SyncLobbyInfo();
+			State = ServerState.GameStarted;
+
+			foreach (var c in conns)
+				foreach (var d in conns)
+					Dispatch(c, d, null, 0x7FFFFFFF, new byte[] { 0xBF });
+
+			Dispatch(null, null, new ServerOrder("StartGame", ""));
+
+			foreach (var t in serverTraits.WithInterface<IStartGame>())
+				t.GameStarted(this);
+
+			// Check TimeOut
+			if (Settings.TimeOut > 10000)
+			{
+				new Timer(state =>
+				{
+					((Timer)state).Dispose();
+					Console.WriteLine("Timeout at {0}.", Settings.TimeOut);
+					Environment.Exit(0);
+				})
+				.Change(Settings.TimeOut, 0);
+			}
+		}
+
+		public void EndGame()
+		{
+			foreach (var t in serverTraits.WithInterface<IEndGame>())
+				t.GameEnded(this);
+		}
+
+		void InterpretServerOrder(Connection conn, ServerOrder so)
 		{
-			return nextPlayerIndex++;
+			switch (so.Name)
+			{
+				case "Command":
+					{
+						var handled = false;
+						foreach (var t in serverTraits.WithInterface<IInterpretCommand>())
+							if (handled = t.InterpretCommand(this, conn, GetClient(conn), so.Data))
+								break;
+
+						if (!handled)
+						{
+							Log.Write("server", "Unknown server command: {0}", so.Data);
+							SendOrderTo(conn, "Message", "Unknown server command: {0}".F(so.Data));
+						}
+
+						break;
+					}
+
+				case "HandshakeResponse":
+					ValidateClient(conn, so.Data);
+					break;
+				case "Chat":
+				case "TeamChat":
+				case "PauseGame":
+					Dispatch(null, conn, so);
+					break;
+				case "Pong":
+					{
+						int pingSent;
+						if (!OpenRA.Exts.TryParseIntegerInvariant(so.Data, out pingSent))
+						{
+							Log.Write("server", "Invalid order pong payload: {0}", so.Data);
+							break;
+						}
+
+						var pingFromClient = LobbyInfo.PingFromClient(GetClient(conn));
+						if (pingFromClient == null)
+							return;
+
+						var history = pingFromClient.LatencyHistory.ToList();
+						history.Add(Game.RunTime - pingSent);
+
+						// Cap ping history at 5 values (25 seconds)
+						if (history.Count > 5)
+							history.RemoveRange(0, history.Count - 5);
+
+						pingFromClient.Latency = history.Sum() / history.Count;
+						pingFromClient.LatencyJitter = (history.Max() - history.Min()) / 2;
+						pingFromClient.LatencyHistory = history.ToArray();
+
+						SyncClientPing();
+
+						break;
+					}
+			}
 		}
 
+		#endregion
+
+		#region Connections
+
+		int nextPlayerIndex = 0;
+		public int ChooseFreePlayerIndex() { return nextPlayerIndex++; }
```

Comment:
Because conventions.

Also because ^ say having random variables floating at random places in a class is bad and class fields should be grouped by modifiers and by type at the beginning of the class.
Also this is not a nice formatting for a method.